app/Http/Controllers/TaskCategoriesController.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+use Illuminate\Support\Facades\Validator;
+
+use App\Models\TaskCategory;
+
+class TaskCategoriesController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $categoryList = TaskCategory::all();
+        return view('TaskCategory.index',compact('categoryList'));
+    }
+
+    public function changeStatus(Request $request) 
+    {
+        $taskCategory = TaskCategory::find($request->id);
+
+        if($taskCategory->status == 0) 
+        {
+            $taskCategory->status = 1;
+        } 
+        else  
+        {
+            $taskCategory->status = 0;
+        }
+
+        $taskCategory->save();
+
+        return redirect()->route('taskcategory.index')->with('success','Status Updated successfully!');
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('TaskCategory.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $validator = Validator::make($request->all(), [
```

Comment:
Move the validation logic to [Form Request](https://laravel.com/docs/8.x/validation#form-request-validation).